src/engine.cc

```diff
@@ -446,14 +455,33 @@ void EngineLoop::CmdUciNewGame() { engine_.NewGame(); }
 void EngineLoop::CmdPosition(const std::string& position,
                              const std::vector<std::string>& moves) {
   std::string fen = position;
-  if (fen.empty()) fen = ChessBoard::kStartposFen;
+  if (fen.empty()) {
+    fen = ChessBoard::kStartposFen;
+  }
   engine_.SetPosition(fen, moves);
 }
 
+void EngineLoop::CmdGetFen() {
+  std::string fen = engine_.GetCurrentPositionFen();
+  return SendResponse(fen);
+}
 void EngineLoop::CmdGo(const GoParams& params) { engine_.Go(params); }
 
 void EngineLoop::CmdPonderHit() { engine_.PonderHit(); }
 
 void EngineLoop::CmdStop() { engine_.Stop(); }
 
+bool ContainsKey(const std::unordered_map<std::string, std::string>& params,
```

Comment:
Instead of duplicating the function, maybe it makes sense to create `utils/misc.h` and move it there? (also use it in uciloop.cc). Something like:

```cpp
template<class T, U>
bool ContainsKey(const T& container, const U& key) {
   return container.find(key) != container.end();
}

Reply:
Good catch; it was dead code. Is now removed.